### ISSUE TITLE
OpenjdkAnnotationLocation: no support for module declarations

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#21162]
Parent Issue #21113

I have read check documentation: https://checkstyle.org/checks/annotation/openjdkannotationlocation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Note: module-info.java is only parseable after #8240, so this is reproduced on that branch.

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="OpenjdkAnnotationLocation"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > oal/module-info.java <<'EOF'
@Deprecated module com.example.app {
    requires java.base;
}
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml oal/module-info.java

Starting audit...
Audit done.
```

The same annotation placement on a class is reported:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > A.java <<'EOF'
package com.example;

@Deprecated class A {
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.10.0-SNAPSHOT-all.jar -c config.xml A.java

Starting audit...
[ERROR] A.java:3:13: Annotations must be on a separate line from 'A'. [OpenjdkAnnotationLocation]
Audit done.
Checkstyle ends with 1 errors.
```

**Describe what you expect in detail.**

A module declaration can carry annotations, and they should follow the same placement rule as annotations on a type declaration, so the module input should be reported. `MODULE_DEF` is not in the Check's acceptable tokens, so the declaration is never visited.

Expected on the module input:

```
1:13: Annotations must be on a separate line from 'com.example.app'.
```

**Additional context**

The Google style equivalent for module annotations is tracked separately in #17172. This issue covers the OpenJDK style Check.

**Comment by romani** [comment]
New token added to Check's available set `TokenTypes.MODULE_DEF`

**Comment by mahfouz72** [comment]
fix is merged at https://github.com/checkstyle/checkstyle/pull/21235
